### PR TITLE
(fleet) e2e improvements

### PR DESCRIPTION
**Pull request by arbll**
Rewrote the e2e setup:
- Now uses the official install script
- Now installs the pipeline agent and installer
- Direct access to the install script as well as installer commands instead of relying on a pre-setup in the other repo
- Made the injector test to be based on the host instead of docker (will add back docker later)
- Added a way to capture the state of an host in a way that is easy to assert, especially for systemd, FS and user/groups.

For now this removes some tests related to the agent but that's fine since it's not the immediate prio.

### CODE REVIEW

File: test/new-e2e/tests/installer/all_packages_test.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+var (
+	amd64Flavors = []e2eos.Descriptor{
+		e2eos.Ubuntu2204,
+		// e2eos.AmazonLinux2,
+		// e2eos.Debian12,
+		// e2eos.RedHat9,
+		// e2eos.Suse15,
+		// e2eos.Fedora37,
+		// e2eos.CentOS7,
```
Comment (BaptisteFoy) [inline test/new-e2e/tests/installer/all_packages_test.go:37]: would be nice to uncomment these

File: test/new-e2e/tests/installer/all_packages_test.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	}
+	for _, flavor := range arm64Flavors {
+		flavor.Architecture = e2eos.ARM64Arch
+		flavors = append(flavors, flavor)
+	}
+	for _, flavor := range flavors {
+		for _, test := range packagesTests {
+			suite := test(flavor, flavor.Architecture)
+			t.Run(suite.Name(), func(t *testing.T) {
+				t.Parallel()
```
Comment (BaptisteFoy) [inline test/new-e2e/tests/installer/all_packages_test.go:65]: **❓ Question:** Isn't t.Run synchronous?‏

File: test/new-e2e/tests/installer/all_packages_test.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+func (s *packageBaseSuite) InstallPackageLatest(pkg string) {
+	env := map[string]string{
+		"DD_API_KEY": apiKey(),
+		"DD_SITE":    "datadoghq.com",
+	}
+	s.Env().RemoteHost.MustExecute(fmt.Sprintf(`sudo -E datadog-installer install oci://gcr.io/datadoghq/%s-package:latest`, strings.TrimPrefix(pkg, "datadog-")), client.WithEnvVariables(env))
+}
+
+func (s *packageBaseSuite) Purge() {
+	s.Env().RemoteHost.MustExecute("sudo apt-get remove -y --purge datadog-installer || sudo yum remove -y datadog-installer")
```
Comment (KevinFairise2) [inline test/new-e2e/tests/installer/all_packages_test.go:199]: You might need to add the equivalent command for `Suse` that would be using `zypper` when you want to run e2e tests on Suse

**Review by KevinFairise2 — APPROVED** [review]

**Review by BaptisteFoy — APPROVED** [review]

File: test/new-e2e/tests/installer/all_packages_test.go (hunk 2 above)
Comment (KevinFairise2) [reply test/new-e2e/tests/installer/all_packages_test.go:65]: It is, unless you add `t.Parallel()`

Comment (BaptisteFoy) [reply test/new-e2e/tests/installer/all_packages_test.go:65]: Noice

**Review by pducolin — APPROVED** [review]

**Comment by pducolin** [comment]
Thank you for the fix, adding a test in a separate PR

File: test/new-e2e/tests/installer/systemd.go [hunk 4]
```diff
@@ -1,90 +0,0 @@
...
-
-func getMonotonicTimestamp(t *testing.T, host *components.RemoteHost) int64 {
-	res := strings.TrimSpace(host.MustExecute("sudo journalctl -n 1 --output=json 2> /dev/null"))
-	var log JournaldLog
-	err := json.Unmarshal([]byte(res), &log)
-	require.NoError(t, err, res)
-	return log.MonotonicTimestamp
-}
-
-// JournaldLog represents a log entry from journald
```
Comment (raphaelgavache) [inline test/new-e2e/tests/installer/systemd.go:29]: why do you remove the experiment test `TestExperimentCrash` ? and those helpers to test experiments

File: test/new-e2e/tests/installer/docker.go [hunk 5]
```diff
@@ -1,132 +0,0 @@
...
-	"time"
-
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
-	"github.com/DataDog/test-infra-definitions/components/os"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-)
-
-// installDocker installs docker on the host
-func installDocker(distro os.Descriptor, arch os.Architecture, t *testing.T, host *components.RemoteHost) {
```
Comment (raphaelgavache) [inline test/new-e2e/tests/installer/docker.go:22]: same for docker, if you remove we'll need to rewrite those tests to cover the injector install section https://github.com/DataDog/datadog-agent/blob/main/pkg/fleet/installer/service/docker.go

File: test/new-e2e/tests/installer/systemd.go (hunk 4 above)
Comment (arbll) [reply test/new-e2e/tests/installer/systemd.go:29]: backported them

**Pull request merged**